receiver/oktareceiver/logs.go

```diff
@@ -0,0 +1,237 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oktareceiver
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/okta/okta-sdk-golang/v2/okta"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type oktaLogsReceiver struct {
+	cfg       Config
+	client    httpClient
+	consumer  consumer.Logs
+	doneChan  chan bool
+	logger    *zap.Logger
+	nextURL   string
+	startTime time.Time
+	wg        *sync.WaitGroup
+}
+
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
```

Comment:
I'd add this function as well that can get called in `Shutdown()`
```suggestion
	Do(req *http.Request) (*http.Response, error)
    CloseIdleConnections()
```
More info: https://pkg.go.dev/net/http#Client.CloseIdleConnections